Tasks/NuGetCommand/nugetpack.ts

```diff
@@ -0,0 +1,189 @@
+import * as tl from "vsts-task-lib/task";
+// Remove once task lib 2.0.4 releases
+global['_vsts_task_lib_loaded'] = true;
+import * as nutil from "nuget-task-common/Utility";
+import nuGetGetter = require("nuget-task-common/NuGetToolGetter");
+import * as path from "path";
+import * as ngToolRunner from "./Common/NuGetToolRunner";
+import * as packUtils from "./Common/NuGetPackUtilities";
+import INuGetCommandOptions from "./Common/INuGetCommandOptions";
+
+class PackOptions implements INuGetCommandOptions {
+    constructor(
+        public nuGetPath: string,
+        public outputDir: string,
+        public includeReferencedProjects: boolean,
+        public version: string,
+        public properties: string[],
+        public verbosity: string,
+        public configFile: string,
+        public environment: ngToolRunner.NuGetEnvironmentSettings
+    ) { }
+}
+
+export async function run(): Promise<void> {
+    nutil.setConsoleCodePage();
+
+    let searchPattern = tl.getPathInput("searchPatternPack", true);
+    let configuration = tl.getInput("configurationToPack");
+    let outputDir = tl.getPathInput("outputDir");
+    let versioningScheme = tl.getInput("versioningScheme");
+    let includeRefProj = tl.getBoolInput("includeReferencedProjects");
+    let versionEnvVar = tl.getInput("versionEnvVar");
+    let majorVersion = tl.getInput("requestedMajorVersion");
+    let minorVersion = tl.getInput("requestedMinorVersion");
+    let patchVersion = tl.getInput("requestedPatchVersion");
+    let propertiesInput = tl.getInput("buildProperties");
+    let verbosity = tl.getInput("verbosityPack");
+    try{
+        if(versioningScheme !== "off" && includeRefProj)
+        {
+            tl.warning(tl.loc("Warning_AutomaticallyVersionReferencedProjects"));
+        }
+
+        let version: string = undefined;
+        switch(versioningScheme)
+        {
+            case "off":
+                break;
+            case "byPrereleaseNumber":
+                tl.debug(`Getting prerelease number`);
+
+                let nowUtcString = packUtils.getUtcDateString();
+                version = `${majorVersion}.${minorVersion}.${patchVersion}-CI-${nowUtcString}`;
+                break;
+            case "byEnvVar":
+                tl.debug(`Getting version from env var: ${versionEnvVar}`);
+                version = tl.getVariable(versionEnvVar);
+                if(!version)
+                {
+                    tl.setResult(tl.TaskResult.Failed, tl.loc("Error_NoValueFoundForEnvVar"));
+                    break;
+                }
+                break;
+            case "byBuildNumber":
+                tl.debug("Getting version number from build number")
+
+                if(tl.getVariable("SYSTEM_HOSTTYPE") === "release")
+                {
+                    tl.setResult(tl.TaskResult.Failed, tl.loc("Error_AutomaticallyVersionReleases"));
```

Comment:
So... while trying to validate this I noticed I cannot even use the task in a release. Investigating.